src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public boolean add(T value) {
-        return false;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(Node<T> previous, T data, Node<T> next) {
+            this.data = data;
+            this.next = next;
+            this.previous = previous;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public boolean add(T data) {
+        insertLast(data);
+        return true;
+    }
 
+    @Override
+    public void add(T data, int index) {
+        checkBounds(index);
+        if (index == size) {
+            insertLast(data);
+        } else {
+            insertBefore(data, getNode(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            insertLast(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return getNode(index).data;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = getNode(index);
+        T oldValue = node.data;
+        node.data = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return removeNode(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+
+        for (Node<T> first = head; first != null; first = first.next) {
+            if (first.data == null) {
+                removeNode(first);
+                return true;
+            }
+            if (object != null && object.equals(first.data)) {
+                removeNode(first);
+                return true;
+            }
```

Comment:
```suggestion
            if (first.data == object || object != null && object.equals(first.data)) {
                removeNode(first);
                return true;
            }
```
let's simplify this check

Reply:
fixed!